### PR TITLE
Update IMV details page

### PR DESCRIPTION
**Pull request by vikram-raj**
**Fixes**: 
https://issues.redhat.com/browse/ODC-4997

**Description**: 
Update IMV details page
1. Add filters for vulnerabilities table
2. Add column sorting in the vulnerabilities table.
3. Add `Manifest` detail item.

**Screen shots / Gifs for design review**: 
Screenshot 2020-11-17 at 4 34 04 PM
Screenshot 2020-11-17 at 4 34 25 PM
Screenshot 2020-11-17 at 4 34 41 PM
Screenshot 2020-11-17 at 4 35 05 PM

**Unit test coverage report**: 
Screenshot 2020-11-17 at 4 42 19 PM

**Test setup:**
Install Quay Container Security operator

### CODE REVIEW

**Comment by itsptk** [comment]
Looks great!

FYI @alecmerdler @dmesser

File: frontend/packages/container-security/src/components/ImageVulnerabilitiesList.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        return _.sortBy(
+          _.flatten(
+            resources?.imageVulnerabilities?.data?.spec.features.map((feature) =>
+              feature.vulnerabilities.map((vulnerability) => ({ feature, vulnerability })),
+            ),
+          ),
+          (v: any) => priorityFor(v.vulnerability.severity).index,
+        );
+      }}
+      namespace={props.match.params.ns}
```
Comment (invincibleJai) [inline frontend/packages/container-security/src/components/ImageVulnerabilitiesList.tsx:52]: can we use destructure value from above

File: frontend/packages/container-security/src/components/ImageVulnerabilityToggleGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import { ToggleGroup, ToggleGroupItem } from '@patternfly/react-core';
+import { totalVulnFor, vulnPriority } from '../const';
+import { ImageManifestVuln } from '../types';
+import {
+  getFixableVulnerabilitiesCount,
+  getVulnerabilityType,
+  VulnerabilitiesType,
+} from './image-vulnerability-utils';
+import './ImageVulnerabilityTogggleGroup.scss';
+import { MsgBox } from '@console/internal/components/utils';
```
Comment (invincibleJai) [inline frontend/packages/container-security/src/components/ImageVulnerabilityToggleGroup.tsx:15]: sort imports

File: frontend/packages/container-security/src/components/ImageVulnerabilityToggleGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        />
+        <ToggleGroupItem
+          buttonId={VulnerabilitiesType.baseImage}
+          isSelected={isSelected === VulnerabilitiesType.baseImage}
+          onChange={handleItemClick}
+          text={t('container-security~Base image')}
+        />
+      </ToggleGroup>
+      {totalSelectedVulnCount > 0 ? (
+        <div style={{ display: 'flex' }}>
```
Comment (karthikjeeyar) [inline frontend/packages/container-security/src/components/ImageVulnerabilityToggleGroup.tsx:73]: Move this style to `ImageVulnerabilityTogggleGroup.scss`

File: frontend/packages/container-security/src/components/image-manifest-vuln.tsx [hunk 4]
```diff
@@ -28,14 +28,15 @@ import {
...
+import { ContainerLink } from '@console/internal/components/pod';
 import { vulnPriority, totalFor, priorityFor } from '../const';
-import { ImageManifestVuln, Feature, Vulnerability } from '../types';
+import { ImageManifestVuln } from '../types';
 import { ImageManifestVulnModel } from '../models';
 import { quayURLFor } from './summary';
+import ImageVulnerabilitiesList from './ImageVulnerabilitiesList';
 import './image-manifest-vuln.scss';
-import { ContainerLink } from '@console/internal/components/pod';
+import ImageVulnerabilityToggleGroup from './ImageVulnerabilityToggleGroup';
```
Comment (karthikjeeyar) [inline frontend/packages/container-security/src/components/image-manifest-vuln.tsx:39]: nit: sort imports

File: frontend/packages/container-security/src/components/image-manifest-vuln.tsx [hunk 5]
```diff
@@ -52,143 +53,35 @@ export const totalCount = (obj: ImageManifestVuln) => {
...
-      </div>
-      <div className="col-lg-2 col-md-2 col-sm-3 hidden-xs">{props.packageName}</div>
-      <div className="col-lg-2 col-md-2 hidden-sm hidden-xs">{props.currentVersion}</div>
-      <div className="col-lg-3 col-md-3 hidden-sm hidden-xs">
-        {props.vulnerability.fixedby || '-'}
-      </div>
+export const ImageManifestVulnDetails: React.FC<ImageManifestVulnDetailsProps> = (props) => (
+  <>
+    <div className="co-m-pane__body">
+      <SectionHeading text="Image Manifest Vuln Details" />
```
Comment (karthikjeeyar) [inline frontend/packages/container-security/src/components/image-manifest-vuln.tsx:59]: localize this text.

File: frontend/packages/container-security/src/components/image-manifest-vuln.tsx [hunk 6]
```diff
@@ -52,143 +53,35 @@ export const totalCount = (obj: ImageManifestVuln) => {
...
-                    </span>
-                  ) : null,
-                )
-                .toArray()}
-            </div>
-          </div>
+        <div className="col-sm-6">
+          <dl className="co-m-pane__details">
+            <DetailsItem label="Registry" obj={props.obj} path="spec.image" />
+            <DetailsItem label="Manifest" obj={props.obj} path="obj.spec.manifest">
```
Comment (karthikjeeyar) [inline frontend/packages/container-security/src/components/image-manifest-vuln.tsx:70]: Localization needed.

**Comment by invincibleJai** [comment]
@vikram-raj we may need to drop point 1 and 3 in description

**Comment by vikram-raj** [comment]
@invincibleJai Removed the `Type` and `Source` column from the vulnerabilities table and removed the group toggle(`App dependency` and `Base image`) for vulnerabilities.

Quay Container Security operator has some issues in the installation. I will update the screenshots once it starts working.

**Comment by vikram-raj** [comment]
@invincibleJai I updated the screenshots as well. PTAL. Thanks

File: frontend/packages/container-security/src/components/ImageVulnerabilitiesList.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    <MultiListPage
+      {...props}
+      resources={[
+        {
+          kind: referenceForModel(ImageManifestVulnModel),
+          namespaced: true,
+          namespace,
+          name,
+          isList: false,
+          prop: 'imageVulnerabilities',
```
Comment (invincibleJai) [inline frontend/packages/container-security/src/components/ImageVulnerabilitiesList.tsx:63]: shall we have `optional: true` as well here?

File: frontend/packages/container-security/src/components/ImageVulnerabilityRow.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  '',
+  '',
+  classNames('pf-m-hidden', 'pf-m-visible-on-md'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-md'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-lg'),
+];
+
+const ImageVulnerabilityRow: RowFunction<ImageVulnerabilityRowProps> = (props) => {
+  const { obj, index, key, style } = props;
+  const { vulnerability, feature } = obj;
```
Comment (invincibleJai) [inline frontend/packages/container-security/src/components/ImageVulnerabilityRow.tsx:24]:
```suggestion
  const { obj : { vulnerability, feature }, index, key, style } = props;
```

File: frontend/packages/container-security/src/components/image-manifest-vuln.tsx [hunk 9]
```diff
@@ -178,13 +112,27 @@ export const ImageManifestVulnDetails: React.FC<ImageManifestVulnDetailsProps> =
...
+                  href={quayURLFor(props.obj)}
+                />
+              </DetailsItem>
             </dl>
           </div>
         </div>
       </div>
-      <div className="co-m-pane__body">
-        <ImageVulnerabilitiesTable features={props.obj.spec.features} />
+      <div className="imagevulnerabilitieslist-wrapper">
```
Comment (invincibleJai) [inline frontend/packages/container-security/src/components/image-manifest-vuln.tsx:134]:
```suggestion
      <div className="imagevulnerabilitieslist__wrapper">
```

can even use pkg name i.e `cs-imagevulnerabilitieslist__wrapper`. WDYT

**Comment by invincibleJai** [comment]
/lgtm

Verified the changes works as expected

**Comment by invincibleJai** [comment]
/lgtm cancel

tests are failing

**Comment by vikram-raj** [comment]
@invincibleJai Fixed the tests. Thanks.

**Pull request merged**